src/components/services/CreateAccountService.jsx

```diff
@@ -1,60 +1,59 @@
 import React from 'react'
+import { translate } from 'cozy-ui/react/I18n'
+
+import AccountConnection from '../../containers/AccountConnection'
+
 import { connect } from 'react-redux'
 
 import {
-  getConnectionStatus,
-  getKonnectorConnectedAccount,
-  getTriggerByKonnector
+  endConnectionCreation,
+  isConnectionRunning,
+  isCreatingConnection,
+  startConnectionCreation
+} from '../../ducks/connections'
+import { isFetchingRegistryKonnector } from '../../ducks/registry'
+import {
+  getCreatedConnectionAccount,
+  getTriggerByKonnectorAndAccount
 } from '../../reducers'
 
-import AccountConnection from '../../containers/AccountConnection'
-
 class CreateAccountService extends React.Component {
   constructor(props, context) {
     super(props, context)
-    const { t, konnector, existingAccount } = props
-    const values =
-      (existingAccount && Object.assign({}, existingAccount.auth)) || {}
-    // Split the actual folderPath account to get namePath & folderPath values
-    if (existingAccount && values.folderPath) {
-      values.folderPath = existingAccount.auth.folderPath.substring(
-        0,
-        existingAccount.auth.folderPath.lastIndexOf('/')
-      )
-      values.namePath = existingAccount.auth.folderPath.substring(
-        existingAccount.auth.folderPath.lastIndexOf('/') + 1,
-        existingAccount.auth.folderPath.length
-      )
-    } else if (
-      (!existingAccount &&
-        konnector.fields &&
+    const { t, konnector } = props
+    const values = {}
+
+    if (
+      (konnector.fields &&
         konnector.fields.advancedFields &&
         konnector.fields.advancedFields.folderPath) ||
-      (!existingAccount && konnector.fields && konnector.folderPath)
+      (konnector.fields && konnector.folderPath)
     ) {
       values.folderPath = t('account.config.default_folder', {
         name: konnector.name
       })
       values.namePath = konnector.name
     }
+
     this.setState({ values: values })
+
+    this.props.startCreation(this.props.konnector)
   }
 
-  componentWillReceiveProps(props) {
-    if (this.props.status === 'running') {
-    } else if (props.status && props.status === 'running') props.onSuccess()
+  onSuccess(account) {
+    this.props.endCreation()
+    this.props.onSuccess(account)
   }
 
   render() {
-    const { onSuccess, alertDeleteSuccess, konnector, trigger } = this.props
+    const { konnector, t } = this.props
     const { values } = this.state
     return (
       <div className="coz-service-content">
         <AccountConnection
-          alertDeleteSuccess={alertDeleteSuccess}
           connector={konnector}
-          onSuccess={onSuccess}
-          trigger={trigger}
+          onNext={account => this.onSuccess(account)}
```

Comment:
I haven't looked at how you usually structure things in cozy-collect, but I'm very fond of:

```
onSuccess = () => {…do things}
render(){
  return <Comp onEvent={this.onSuccess} />
}
```

It has the advantage of not needing to pass the arguments around, nor creating a new function to preserve the context.

edit: nevermind, I've seen the synthax you use here is used everywhere else too, so better keep it that way.

Reply:
By the way I am interested, how do you keep the context ?

Reply:
It's because of the fat arrow (`=>`)! 

`onSuccess = () => {…do things}` and `onSuccess(){…do things}` both create a prop called `onSuccess` which is a function, but using `=>` in the declaration forcefully binds the function to `this`, which happens to be the class instance!

Reply:
Great, thanks !

Reply:
For consistency we'll keep the existing pattern for now, but we will soon make a refactoring PR to fix all this callback properties. 👌